Dockerfile

```diff
@@ -13,8 +13,10 @@ RUN go mod download
 COPY main.go main.go
 COPY pkg/ pkg/
 
+ARG TARGETARCH
+
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH="${TARGETARCH}" go build -a -o manager main.go
 
 FROM registry.access.redhat.com/ubi8/ubi-minimal
```

Comment:
Which image is pulled here? Do the resulting image arch and the BUILDPLATFORM match?

Reply:
I have tested the whole workflow on my own fork against ghcr.io.
The BUILDPLATFORM means that the first stage is run natively, e.g. on amd64, while the TARGETARCH may be s390x.
The 2nd stage is not affected, so if you target s390x, you compile the s390x binary on amd64 via cross-compile but still get a s390x image in the 2nd stage.

Reply:
I believe we need to update the following:
FROM registry.access.redhat.com/ubi8/ubi-minimal
to:
FROM --platform=linux/${TARGETARCH} registry.access.redhat.com/ubi8/ubi-minimal
This is necessary because FROM registry.access.redhat.com/ubi8/ubi-minimal works fine with docker buildx, but it fails with podman when building for multiple platforms.

Reply:
That's what I had in mind.

Reply:
Thanks
@jschintag 
would you able to take a look please?

Generally speaking, will be good that we check all the PRs in the effort locally with both docker and podman please

Reply:
@oshoval I have tested local builds with both podman and docker. Meaning s390x image build on s390x machine.

I have also tried adding the `--platform` arg to podman, however the result seems to be that it is mostly ignored. Might be that i need to add some extra args or so?

Reply:
i am not familiar about it by heart tbh
maybe lets wait please from ideas from the people involved

curious, iiuc it works without it on all the required combinations ? 
native and s390x (cross / native) are most important, on podman mainly
to see it builds and runs

Reply:
I have tested native with docker/podman by running `make build` on s390x. Let me test it on x86 as well just to be sure, though there should not be a difference. Cross as i said seemed to not work for me with OCI: podman, host: s390x, target: amd64

Reply:
Native amd64 -> amd64 works as well with docker/podman.

Reply:
> host: s390x, target: amd64 

we can fix this one later specifically according needs, because atm native host/target amd64 works which is what important

host: amd64, target: s390x - does work ? (podman / docker, higher priority for podman)

maybe the PRs of @ashokpariya0 might help in this, if it works there

Thanks

Reply:
The combination of host: s390x and target: amd64 is not supported, so there is no need to test this configuration.
As @oshoval pointed out we need to test following:

Host: amd64
Target: amd64, arm64, and s390x.

Also For the final stage of the build, better to use the following:
FROM --platform=linux/${TARGETARCH} registry.access.redhat.com/ubi8/ubi-minimal
This approach is necessary when building multi-platform images using Podman. You will need to create a Podman manifest and then add the individual platform-specific builds one by one to the manifest. Also adding (linux/${TARGETARCH}) works well with docker buildx.